client/components/cloud/project/billing/additional-services/additional-services.component.controller.js

```diff
@@ -0,0 +1,90 @@
+class AdditionalServicesComponentCtrl {
+  constructor($stateParams, $translate, CucControllerHelper,
+    OvhApiMe, CucServiceHelper, OvhApiCloudProjectInstance) {
+    this.$stateParams = $stateParams;
+    this.$translate = $translate;
+    this.CucControllerHelper = CucControllerHelper;
+    this.OvhApiMe = OvhApiMe;
+    this.CucServiceHelper = CucServiceHelper;
+    this.OvhApiCloudProjectInstance = OvhApiCloudProjectInstance;
+  }
+
+  $onInit() {
+    this.initAccordians();
+    if (this.getPlatformCount() > 0) {
+      this.initInstances();
+    }
+  }
+
+  /**
+   * initiate UI accordian state, one accordian per platform
+   */
+  initAccordians() {
+    const accordions = {};
+    _.forEach(_.keys(this.services), (key) => {
+      accordions[key] = false;
+    });
+    this.toggle = {
+      accordions,
+    };
+  }
+
+  /**
+   * get all instances for this PCI project
+   */
+  initInstances() {
+    // instances details are required to get instances name
+    this.instances = this.CucControllerHelper.request.getHashLoader({
+      loaderFunction: () => this.OvhApiCloudProjectInstance.v6()
+        .query({
+          serviceName: this.$stateParams.projectId,
+        })
+        .$promise.then((instances) => {
+          _.forEach(_.keys(this.services), (key) => {
+            _.map(this.services[key], (service) => {
+              const instance = _.find(instances, { id: service.instanceId });
+              if (instance) {
+                _.set(service, 'name', instance.name || service.instanceId);
+              } else {
+                // instance deleted, name not available, show the id
+                _.set(service, 'name', service.instanceId);
+              }
+            });
+          });
+        }),
+    });
+    return this.instances.load();
+  }
+
+  /**
+   * returns number of platforms (like bigdat platform) deployed on this PCI
+   */
+  getPlatformCount() {
+    return _.keys(this.services).length;
```

Comment:
Is `size` make sense?

```suggestion
    return _.size(this.services);
```